app/controllers/decidim/action_delegator/admin/invite_participants_controller.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+module Decidim
+  module ActionDelegator
+    module Admin
+      class InviteParticipantsController < ActionDelegator::Admin::ApplicationController
+        include NeedsPermission
+
+        helper_method :current_setting, :users_list_to_invite, :participant, :form
+
+        def invite_user
+          enforce_permission_to :invite, :participant, resource: current_setting
+
+          Decidim::InviteUser.call(form) do
+            on(:ok) do
+              notice = t("invite_user.success", scope: "decidim.action_delegator.admin.invite_participants")
+              redirect_to decidim_admin_action_delegator.setting_participants_path(current_setting), notice: notice
+            end
+          end
+        end
+
+        def invite_all_users
+          enforce_permission_to :invite, :participant, resource: current_setting
+
+          InviteParticipantsJob.perform_later(users_list_to_invite.to_a, current_organization)
```

Comment:
```suggestion
          InviteParticipantsJob.perform_later(current_setting.id, current_organization)
```

In here, converting the list of users to an array will work, but if the list of users is big, it can cause memory problems. Alternatively, we can pass only the setting id and move the method `invite_all_users` to the job class.
Once there, let's iterate in this manner, also for avoiding big results memory issues:

```ruby
users_list_to_invite.find_each do |participant|
...
```